src/mongo_client.ts

```diff
@@ -638,6 +640,20 @@ export class MongoClient extends TypedEventEmitter<MongoClientEvents> implements
    * @param force - Force close, emitting no events
    */
   async close(force = false): Promise<void> {
+    if (this.closeLock) {
+      return await this.closeLock;
+    }
+
+    try {
+      this.closeLock = this._close(force);
+      await this.closeLock;
+    } finally {
+      // release
+      this.closeLock = undefined;
+    }
+  }
+
+  async _close(force = false): Promise<void> {
```

Comment:
```suggestion
  /** @internal */
  async _close(force = false): Promise<void> {
```

Can we make this internal?  We could also make it TS private too, if you want to.